libevmasm/SemanticInformation.cpp

```diff
@@ -97,6 +114,7 @@ std::vector<SemanticInformation::Operation> SemanticInformation::readWriteOperat
 	{
 		assertThrow(memory(_instruction) == Effect::Write, OptimizerException, "");
 		assertThrow(storage(_instruction) == Effect::None, OptimizerException, "");
+		assertThrow(transientStorage(_instruction) == Effect::None, OptimizerException, "");
```

Comment:
any particular reason this `assertThrow` for `transientStorage(_instruction)`  is missing from the case:

```cpp
case Instruction::MCOPY:
{
		assertThrow(memory(_instruction) != Effect::None, OptimizerException, "");
		assertThrow(storage(_instruction) == Effect::None, OptimizerException, "");
```

Reply:
PRs for Cancun opcodes were done in parallel. While the one for MCOPY was being written, the transient storage effects were not in the code base yet, so it could not assert this at that point. TSTORE PR could not assert it either because in it MCOPY did not exist.

We could have gone back to this after transient storage PR was merged but this is such a small detail that no one remembered to do that (it's just an extra sanity check, pretty unlikely to fail).

Again, you're right that it should technically be there so we'd accept a PR correcting that.

Reply:
- https://github.com/ethereum/solidity/pull/14920